src/main/java/bio/terra/cloudfunctions/common/App.java

```diff
@@ -0,0 +1,32 @@
+package bio.terra.cloudfunctions.common;
+
+import bio.terra.cloudevents.v1.CloudEventType;
+
+/**
+ * This abstract class represents the business application that interprets CloudEvent messages.
+ *
+ * <p>Sub-classes overrides the process() method with business logic to achieve desired functional
+ * goals.
+ *
+ * <p>Sub-classes can be integrated with a DI framework to deploy the business logic like as a
+ * Service.
+ */
+public abstract class App {
+  private CloudEventType cloudEventType;
+  private Object message;
```

Comment:
I'd like to use our own message abstraction instead of an Object and not use io.cloudevents.  Something like 
```
public abstract class MessageProcessor {

  protected final FileMessage message;

  public MessageProcessor(FileMessage message) {
    this.message = message;
  }
  
  public abstract void processMessage();
}
```

Reply:
I am fine with this. Consider the following changes as well:

- rename `App -> StorageObjectEventApp`
- change `FileMessage -> StorageObjectData` (why not `StorageObjectData` since it has all the properties in `FileMessage`?)